### PR TITLE
[Transform] support flex tensor

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Add pad template and update chain function to support flexible tensor stream.

TODO: Now max rank supported in tensor-transform is 4.
Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: tests/nnstreamer_plugins/unittest_plugins.cc [hunk 1]
```diff
@@ -1909,6 +1909,100 @@ TEST (testTensorTransform, arithmeticChangeOptionString)
   gst_harness_teardown (h);
 }
 
+/**
+ * @brief Test for tensor_transform arithmetic (flex tensor)
```
Comment (gichan-jang) [inline tests/nnstreamer_plugins/unittest_plugins.cc:1913]: I think static tensor -> flex tensor transform test is needed to increase the code coverage.

Comment (jaeyun-jung) [reply tests/nnstreamer_plugins/unittest_plugins.cc:1913]: later work : transform-caps in tensor-transform
When called transform-caps(), it should append all possible caps as result, but now only one possible caps supported.

**Review by gichan-jang — APPROVED** [review]
LGTM

File: gst/nnstreamer/tensor_transform/tensor_transform.c [hunk 2]
```diff
@@ -933,33 +933,151 @@ gst_tensor_transform_finalize (GObject * object)
...
+ * @brief Dimension conversion calculation
+ * @param[in] filter "this" pointer
+ * @param[in] direction GST_PAD_SINK if input->output conv
+ * @param[in] idx index of the input tensors
+ * @param[in] in_info tensor info structure of source tensor (input if direction is SINK)
+ * @param[out] out_info tensor info structure of destination tensor (output if direction is SINK)
+ * @return TRUE if success
+ */
+static gboolean
+gst_tensor_transform_convert_dimension (GstTensorTransform * filter,
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_transform/tensor_transform.c:946]: Can you keep the position of this function and only add function declaration to reduce changes? (making reviews easier)

File: tests/nnstreamer_plugins/unittest_plugins.cc [hunk 3]
```diff
@@ -1909,6 +1909,100 @@ TEST (testTensorTransform, arithmeticChangeOptionString)
...
+ */
+TEST (testTensorTransform, arithmeticFlexTensor)
```
Comment (myungjoo) [inline tests/nnstreamer_plugins/unittest_plugins.cc:1915]: negative cases?

File: gst/nnstreamer/tensor_transform/tensor_transform.c (hunk 2 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_transform/tensor_transform.c]: updated

File: tests/nnstreamer_plugins/unittest_plugins.cc (hunk 3 above)
Comment (jaeyun-jung) [reply tests/nnstreamer_plugins/unittest_plugins.cc:1933]: After changing caps string (about tensor format)
All elements with flex caps should have formal condition to check data size. Now tensor-transform does not compare data size in transform().

Comment (myungjoo) [reply tests/nnstreamer_plugins/unittest_plugins.cc:1933]: Ok. Let's postpone it until then.

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**